core/unit_test/CMakeLists.txt

```diff
@@ -24,200 +24,108 @@ KOKKOS_TARGET_INCLUDE_DIRECTORIES(kokkos_gtest PUBLIC ${GTEST_SOURCE_DIR})
-      serial/TestSerial_ViewMapping_subview.cpp
-      serial/TestSerial_ViewOfClass.cpp
-      serial/TestSerial_Crs.cpp
-      serial/TestSerial_WorkGraph.cpp
-      serial/TestSerial_ViewResize.cpp
+    ${Serial_SOURCES}
+    serial/TestSerial_Task.cpp
   )
 ENDIF()
 
 IF(Kokkos_ENABLE_PTHREAD)
   KOKKOS_ADD_EXECUTABLE_AND_TEST(
     UnitTest_Threads
-    SOURCES
-      UnitTestMainInit.cpp
-      threads/TestThreads_AtomicOperations_int.cpp
-      threads/TestThreads_AtomicOperations_unsignedint.cpp
-      threads/TestThreads_AtomicOperations_longint.cpp
-      threads/TestThreads_AtomicOperations_unsignedlongint.cpp
-      threads/TestThreads_AtomicOperations_longlongint.cpp
-      threads/TestThreads_AtomicOperations_double.cpp
-      threads/TestThreads_AtomicOperations_float.cpp
-      threads/TestThreads_AtomicOperations_complexdouble.cpp
-      threads/TestThreads_AtomicOperations_complexfloat.cpp
-      threads/TestThreads_AtomicViews.cpp
-      threads/TestThreads_Atomics.cpp
-      threads/TestThreads_Complex.cpp
-      threads/TestThreads_Init.cpp
-      threads/TestThreads_MDRange_a.cpp
-      threads/TestThreads_MDRange_b.cpp
-      threads/TestThreads_MDRange_c.cpp
-      threads/TestThreads_MDRange_d.cpp
-      threads/TestThreads_MDRange_e.cpp
-      threads/TestThreads_Other.cpp
-      threads/TestThreads_RangePolicy.cpp
-      threads/TestThreads_Reductions.cpp
-      threads/TestThreads_Reducers_a.cpp
-      threads/TestThreads_Reducers_b.cpp
-      threads/TestThreads_Reducers_c.cpp
-      threads/TestThreads_Reducers_d.cpp
-      threads/TestThreads_Scan.cpp
-      threads/TestThreads_SharedAlloc.cpp
-      threads/TestThreads_SubView_a.cpp
-      threads/TestThreads_SubView_b.cpp
-      threads/TestThreads_SubView_c01.cpp
-      threads/TestThreads_SubView_c02.cpp
-      threads/TestThreads_SubView_c03.cpp
-      threads/TestThreads_SubView_c04.cpp
-      threads/TestThreads_SubView_c05.cpp
-      threads/TestThreads_SubView_c06.cpp
-      threads/TestThreads_SubView_c07.cpp
-      threads/TestThreads_SubView_c08.cpp
-      threads/TestThreads_SubView_c09.cpp
-      threads/TestThreads_SubView_c10.cpp
-      threads/TestThreads_SubView_c11.cpp
-      threads/TestThreads_SubView_c12.cpp
-      threads/TestThreads_SubView_c13.cpp
-      threads/TestThreads_Team.cpp
-      threads/TestThreads_TeamReductionScan.cpp
-      threads/TestThreads_TeamScratch.cpp
-      threads/TestThreads_ViewAPI_a.cpp
-      threads/TestThreads_ViewAPI_b.cpp
-      threads/TestThreads_ViewAPI_c.cpp
-      threads/TestThreads_ViewAPI_d.cpp
-      threads/TestThreads_ViewAPI_e.cpp
-      threads/TestThreads_ViewMapping_a.cpp
-      threads/TestThreads_ViewMapping_b.cpp
-      threads/TestThreads_ViewMapping_subview.cpp
-      threads/TestThreads_ViewOfClass.cpp
-      threads/TestThreads_Crs.cpp
-      threads/TestThreads_WorkGraph.cpp
-      threads/TestThreads_ViewResize.cpp
+    SOURCES ${Threads_SOURCES}
```

Comment:
@dhollman Does the task test not work with the ptrhreads backend?

Reply:
Building the test would currently fail with
```
[ 99%] Building CXX object core/unit_test/CMakeFiles/KokkosCore_UnitTest_Threads.dir/threads/TestThreads_Task.cpp.o
In file included from kokkos/core/src/Kokkos_Core.hpp:94:0,
                 from kokkos/core/unit_test/TestTaskScheduler.hpp:49,
                 from kokkos/core/unit_test/threads/TestThreads_Task.cpp:46:
kokkos/core/src/Kokkos_TaskScheduler.hpp: In instantiation of ‘class Kokkos::BasicTaskScheduler<Kokkos::Threads, Kokkos::Impl::TaskQueue<Kokkos::Threads, Kokkos::HostSpace> >’:
kokkos/core/src/Kokkos_Future.hpp:285:62:   required from ‘class Kokkos::BasicFuture<long int, Kokkos::BasicTaskScheduler<Kokkos::Threads, Kokkos::Impl::TaskQueue<Kokkos::Threads, Kokkos::HostSpace> > >’
kokkos/core/unit_test/TestTaskScheduler.hpp:82:15:   required from ‘struct TestTaskScheduler::TestFib<Kokkos::BasicTaskScheduler<Kokkos::Threads, Kokkos::Impl::TaskQueue<Kokkos::Threads, Kokkos::HostSpace> > >’
kokkos/core/unit_test/TestTaskScheduler_single.hpp:50:47:   required from here
kokkos/core/src/Kokkos_TaskScheduler.hpp:88:67: error: no type named ‘member_type’ in ‘struct Kokkos::Impl::TaskQueueSpecialization<Kokkos::BasicTaskScheduler<Kokkos::Threads, Kokkos::Impl::TaskQueue<Kokkos::Threads, Kokkos::HostSpace> > >’
   using member_type         = typename specialization::member_type;
```

Reply:
Tasking is not implemented for the pthreads backend AFAIK